src/SignUpPage.tsx

```diff
@@ -0,0 +1,200 @@
+import React, { useState } from 'react';
+import logo from './logo.svg';
+import styled from 'styled-components';
+
+const LoginPageContainer = styled.div`
+	min-height: 100vh;
+	display: flex;
+	flex-direction: column;
+	align-items: center;
+	justify-content: center;
+	background-color: #e5e5e5;
+`;
+
+const LoginContents = styled.div`
+	width: 250px;
+	display: flex;
+	flex-direction: column;
+	justify-content: flex-start;
+	align-items: center;
+`;
+
+const PassrLogo = styled.div`
+	width: 100%;
+`;
+
+const InputContainer = styled.div`
+	display: flex;
+	flex-direction: column;
+	align-items: flex-start;
+	justify-content: center;
+	margin: 1.5em 0;
+	width: 100%;
+`;
+
+const LoginButton = styled.button`
+	width: 100%;
+	margin: 1.5em 0;
+`;
+
+const InputLabel = styled.div`
+	font-weight: 500;
+	font-size: 1.2rem;
+	color: #4961e1;
+`;
+
+const InputBox = styled.input`
+	width: 100%;
+`;
+
+const Links = styled.div`
+	font-size: 0.8rem;
+	font-weight: 500;
+	display: flex;
+	flex-direction: column;
+`;
+
+function validEmail(email: string) {
+	return !!email.match(/.+@.+\..{2,}/);
+}
+
+function compareEmail(email1: string, email2: string) {
+	return !!email1.match(email2);
+}
+
+// to do
+function validPass(password: string) {
+	return !!password.match(/.+@.+\..{2,}/);
+}
+
+function comparePass(password1: string, password2: string) {
+	return !!password1.match(password2);
+}
+
+function SignUpPage() {
+	const [email, setEmail] = useState('');
+	const [confirmEmail, setConfirmEmail] = useState('');
+	const [username, setUsername] = useState('');
+	const [emailErr, setEmailErr] = useState(false);
+	const [userPass, setUserPass] = useState('');
+	const [confirmUserPass, setConfirmUserPass] = useState('');
+	const [passErr, setPassErr] = useState(false);
```

Comment:
This is marked as unused, which means the passErr isn't stopping anything. We should add this to the button disable clause below. 
```
disabled={emailErr || passErr}
```